landoapi/workers/landing_worker.py

```diff
@@ -0,0 +1,448 @@
+        # revisions are marked as ready. In the future this will need to also account for
+        # merge conflicts within the context of a stack.
+
+        if repo.use_revision_worker and job.has_non_ready_revisions():
+            job.transition_status(
+                LandingJobAction.DEFER,
+                message=f"{job} has non ready revisions - retrying later.",
+                commit=True,
+                db=db,
+            )
+            return False
+
+        with hgrepo.for_push(job.requester_email):
+            try:
+                hgrepo.update_repo(repo.pull_path)
+            except Exception as e:
+                message = f"Unexpected error while fetching repo from {repo.pull_path}."
+                logger.exception(message)
+                job.transition_status(
+                    LandingJobAction.FAIL,
+                    message=message + f"\n{e}",
+                    commit=True,
+                    db=db,
+                )
+                self.notify_user_of_landing_failure(job)
+                return True
+
+            # Load all patches.
+            patch_bufs = []
+            for revision in job.get_revisions():
+                patch = revision.patch.encode("utf-8")
+                if not revision.verify_patch_hash(patch):
+                    message = "Aborting, patch has changed since landing trigger."
+                    logger.error(message)
+                    job.transition_status(
+                        LandingJobAction.FAIL,
+                        message=message,
+                        commit=True,
+                        db=db,
+                    )
+                    self.notify_user_of_landing_failure(job)
+                    job.fail_revisions()
+                    # TODO makes sure that repos that do not use
+                    # revision worker will force-update patch on
+                    # next request.
+                    return True
+                patch_bufs.append((revision, patch))
+
+            for revision, patch in patch_bufs:
+                try:
+                    hgrepo.apply_patch(BytesIO(patch))
+                except PatchConflict as exc:
+                    breakdown = self.process_merge_conflict(exc, repo, hgrepo, revision)
+                    message = (
+                        f"Problem while applying patch in revision {revision.revision_id}:\n\n"
+                        f"{str(exc)}"
+                    )
+                    job.error_breakdown = breakdown
+
+                    job.transition_status(
+                        LandingJobAction.FAIL, message=message, commit=True, db=db
+                    )
+                    self.notify_user_of_landing_failure(job)
+                    job.fail_revisions()
+                    db.session.commit()
+                    return True
+                except Exception as e:
+                    # verify below line
+                    if e is NoDiffStartLine:
```

Comment:
I think you want `isinstance` here, but I'd still vote that we catch `NoDiffStartLine` explicitly and handle any repeated behaviour with a function.